flask_resty/view.py

```diff
@@ -784,8 +784,8 @@ def add_item(self, item):
 
         :param object item: The item to add.
         """
-        self.add_item_raw(item)
         self.authorization.authorize_save_item(item)
```

Comment:
Shouldn't we do the auth check first? In practice not too big an issue as the session doesn't commit and rolls back, but this seems odd...

Reply:
my guess is that the intention is to authorize just before calling db.session.commit().

have you run into the use case where this call order matters? 👀

Reply:
> have you run into the use case where this call order matters? 👀

Only noticed it while doing somewhat unrelated Sqlalchemy 2 upgrade work on our end. I don't think it matters in practice, but it does mean that it adds the model into the session even if the caller is unauthorized temporarily

Reply:
yeah, i agree that the current behavior is a bit odd. i've no objection to merging this